### ISSUE TITLE
Certain dynamic routes resolve to unexpected URLs.

### ISSUE THREAD

**Issue by Collinbrown95** [doccano/doccano#992]
I am not sure if this is intended behaviour or an error, but it seems strange to me so I thought that I would flag it.

It appears that certain dynamic routes resolve to unexpected URLs. For example, ```127.0.0.1:3000/projects/1/labels``` resolves to ```127.0.0.1:3000/projects/1``` when entered directly in the address bar of the browser.

This behaviour leads to a specific issue where the localized ```nuxt-link``` (see the [localization pull request](https://github.com/doccano/doccano/pull/976)) for project pages brings the user to the default locale of that project page on their first visit (but not on subsequent visits).

Discussion
-----------------

If this is intended behaviour, need to diagnose why it causes the localization error mentioned above.

If this is unintended behaviour, then need to diagnose why it is happening and fix it.

**Comment by Hironsan** [comment]
This is not intended behavior.

The problem is in the [check-admin.js](https://github.com/doccano/doccano/blob/master/frontend/middleware/check-admin.js) middleware. It decides whether to redirect or not, depending on the user's role. The user's role value is in the Vuex store, but if we entered the address bar of the browser directly, the value is disappeared. As a result, the middleware redirects the user to `projects/{project_id}` page.

One way to solve the problem is to check if the user's role is empty before deciding whether to redirect. If empty,  the middleware fetches the role. The following code is a new `check-admin.js`:

```
export default async function({ store, route, redirect }) {
  if (store.getters['projects/isEmpty']) {
    await store.dispatch('projects/setCurrentProject', route.params.id)
  }
  const role = store.getters['projects/getCurrentUserRole']
  const projectRoot = '/projects/' + route.params.id
  const path = route.fullPath.replace(/\/$/g, '')
  if (!role.is_project_admin && path !== projectRoot) {
    return redirect(projectRoot)
  }
}
```

**Comment by Hironsan** [comment]
Resolved in #976
